lib/pegboard/manager/src/container.rs

```diff
@@ -0,0 +1,131 @@
+use std::{
+	os::unix::process::CommandExt,
+	path::PathBuf,
+	process::{Command, Stdio},
+};
+
+use anyhow::*;
+use nix::{
+	sys::wait::{waitpid, WaitStatus},
+	unistd::{fork, ForkResult},
+};
+use pegboard::protocol;
+use url::Url;
+use uuid::Uuid;
+
+use crate::{ctx::Ctx, utils};
+
+const VECTOR_SOCKET_ADDR: &str = "127.0.0.1:5021";
+
+pub struct Container {
+	pub container_id: Uuid,
+	pub image_artifact_url: String,
+	pub container_runner_binary_url: String,
+	pub root_user_enabled: bool,
+	pub stakeholder: protocol::Stakeholder,
+}
+
+impl Container {
+	pub fn new(
+		container_id: Uuid,
+		image_artifact_url: String,
+		container_runner_binary_url: String,
+		stakeholder: protocol::Stakeholder,
+	) -> Self {
+		Container {
+			container_id,
+			image_artifact_url,
+			container_runner_binary_url,
+			root_user_enabled: todo!(),
+			stakeholder,
+		}
+	}
+
+	pub async fn start(self, ctx: &Ctx) -> Result<()> {
+		// Get local path of container runner
+		let container_runner_path = ctx
+			.fetch_container_runner(&self.container_runner_binary_url)
+			.await?;
+
+		let url = Url::parse(&self.image_artifact_url)?;
+		let path_stub = utils::get_s3_path_stub(&url, false)?;
+		let path = ctx.container_path(self.container_id).join(path_stub);
+
+		// Download container image
+		utils::download_file(&self.image_artifact_url, &path).await?;
+
+		// Spawn container
+		self.spawn(&ctx, container_runner_path)
+	}
+
+	fn spawn(&self, ctx: &Ctx, container_runner_path: PathBuf) -> Result<()> {
+		let container_path = ctx.container_path(self.container_id);
+
+		let mut env = vec![
+			(
```

Comment:
we don't need the `*_META_*` and lowercase stuff since we no longer use nomad. this was an arfifact of using nomad's meta functionality.